runelite-client/src/main/java/net/runelite/client/plugins/profiles/ProfilesPanel.java

```diff
@@ -0,0 +1,222 @@
+		txtAccountLabel.addFocusListener(new FocusListener()
+		{
+			@Override
+			public void focusGained(FocusEvent e)
+			{
+				if (txtAccountLabel.getText().equals(ACCOUNT_LABEL))
+				{
+					txtAccountLabel.setText("");
+					txtAccountLabel.setForeground(ColorScheme.LIGHT_GRAY_COLOR);
+				}
+			}
+
+			@Override
+			public void focusLost(FocusEvent e)
+			{
+				if (txtAccountLabel.getText().isEmpty())
+				{
+					txtAccountLabel.setForeground(ColorScheme.MEDIUM_GRAY_COLOR);
+					txtAccountLabel.setText(ACCOUNT_LABEL);
+				}
+			}
+		});
+
+		add(txtAccountLabel, c);
+		c.gridy++;
+
+		// Do not hide username characters until they focus or if in streamer mode
+		txtAccountLogin.setEchoChar((char) 0);
+		txtAccountLogin.setPreferredSize(PREFERRED_SIZE);
+		txtAccountLogin.setForeground(ColorScheme.MEDIUM_GRAY_COLOR);
+		txtAccountLogin.setBackground(ColorScheme.DARKER_GRAY_COLOR);
+		txtAccountLogin.setMinimumSize(MINIMUM_SIZE);
+		txtAccountLogin.addFocusListener(new FocusListener()
+		{
+			@Override
+			public void focusGained(FocusEvent e)
+			{
+				if (ACCOUNT_USERNAME.equals(String.valueOf(txtAccountLogin.getPassword())))
+				{
+					txtAccountLogin.setText("");
+					if (config.isStreamerMode())
+					{
+						txtAccountLogin.setEchoChar('*');
+					}
+					txtAccountLogin.setForeground(ColorScheme.LIGHT_GRAY_COLOR);
+				}
+			}
+
+			@Override
+			public void focusLost(FocusEvent e)
+			{
+				if (txtAccountLogin.getPassword().length == 0)
+				{
+					txtAccountLogin.setForeground(ColorScheme.MEDIUM_GRAY_COLOR);
+					txtAccountLogin.setText(ACCOUNT_USERNAME);
+					txtAccountLogin.setEchoChar((char) 0);
+				}
+			}
+		});
+
+		add(txtAccountLogin, c);
+		c.gridy++;
+		c.insets = new Insets(0, 0, 15, 0);
+
+		JButton btnAddAccount = new JButton("Add Account");
+		btnAddAccount.setPreferredSize(PREFERRED_SIZE);
+		btnAddAccount.setBackground(ColorScheme.DARKER_GRAY_COLOR);
+		btnAddAccount.setMinimumSize(MINIMUM_SIZE);
+		btnAddAccount.addActionListener(e ->
+		{
+			String data = txtAccountLabel.getText() + ":" + String.valueOf(txtAccountLogin.getPassword());
+			log.info(data);
+			this.addAccount(data);
+
+			addProfile(data);
+
+			txtAccountLabel.setText(ACCOUNT_LABEL);
+			txtAccountLabel.setForeground(ColorScheme.LIGHT_GRAY_COLOR);
+
+			txtAccountLogin.setText(ACCOUNT_USERNAME);
+			txtAccountLogin.setEchoChar((char) 0);
+			txtAccountLogin.setForeground(ColorScheme.LIGHT_GRAY_COLOR);
+		});
+
+		add(btnAddAccount, c);
+
+		addAccounts(config.profilesData());
+	}
+
+	void remove(ProfilePanel panel)
+	{
+		super.remove(panel);
+		revalidate();
+		repaint();
+	}
+
+	void redrawProfiles()
+	{
+		profiles.forEach(this::remove);
+		addAccounts(profilesConfig.profilesData());
+	}
+
+	private void addAccount(String data)
+	{
+		c.gridy++;
+		c.insets = new Insets(0, 0, 5, 0);
+		log.info(data);
+		ProfilePanel profile = new ProfilePanel(client, data, profilesConfig);
+		add(profile, c);
+		profiles.add(profile);
```

Comment:
You never clear `profiles` and thus whenever call `redrawProfiles()` it creates new panels and adds them to this list.

You can just move all the profiles into a JPanel and remove this entire list.